src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java

```diff
@@ -398,17 +445,45 @@ private IAIState advanceLadder(final IAIState state)
             return state;
         }
 
+        @NotNull final BlockPos nextLadder =
+                new BlockPos(getOwnBuilding().getLadderLocation().getX(), getLastLadder(getOwnBuilding().getLadderLocation(), world) - 1, getOwnBuilding().getLadderLocation().getZ());
+        final int xOffset = SHAFT_RADIUS * getOwnBuilding().getVectorX();
+        final int zOffset = SHAFT_RADIUS * getOwnBuilding().getVectorZ();
+        boolean repair = true;
+
+        if (world.getBlockState(nextLadder).getBlock() instanceof AirBlock)
```

Comment:
might make sense to put this in a utility functin that returns true if repair and else false

Reply:
yes, but most of this stuff will get removed in the next commit.

Reply:
as it checked the wrong stuff

Reply:
This should be solved now

Reply:
ok now i think i understand what you meant.
1. i thought this was the other check that looks the same
2. yes, i will probably do that.

Reply:
ok i did this.